Dalamud/Interface/Internal/Windows/SettingsWindow.cs

```diff
@@ -561,6 +565,34 @@ private void DrawExperimentalTab()
             var configuration = Service<DalamudConfiguration>.Get();
             var pluginManager = Service<PluginManager>.Get();
 
+            var useCustomPluginWaitBeforeFree = this.pluginWaitBeforeFree.HasValue;
+            if (ImGui.Checkbox(
```

Comment:
I'm a little wary of having this customizable... It has no meaning to users, ideally we'd have a default value that always works. What's the actual problem here - why can wait time affect disposes? Back before the async workaround, we'd just call dispose synchronously on every plugin, and it worked fine.

Reply:
It also used to wait for quite a long time, and that usually gave Tasks ran by plugins to finish. Without this wait time, tasks would keep running even after successful plugin/loader.Dispose, and TE would crash the game when this is followed by Dalamud unload.

Reply:
Fair enough, I guess this is where implementing our own task scheduler would be useful, but that's a topic for another day